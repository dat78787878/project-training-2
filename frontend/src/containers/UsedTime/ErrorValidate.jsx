import React from 'react';

const ErrorValidate = ({ errors }) => {
  return (
    <div className="modal-error">
      {errors.map((error) => {
        return <p key={error}>Error: {error}</p>;
      })}
    </div>
  );
};

export default ErrorValidate;
