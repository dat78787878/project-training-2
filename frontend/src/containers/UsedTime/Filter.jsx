import React from 'react';
import { Col, Form, Row } from 'react-bootstrap';
import { v4 as uuidv4 } from 'uuid';

const Filter = ({
  setDataRender,
  usedTimeData,
  search,
  date,
  type,
  setSearch,
  setDate,
  setType
}) => {
  const listOS = ['Android', 'iOS'];

  const handleSearch = (e) => {
    e.preventDefault();
    const keyFilter = e.target.value.trim().toLowerCase();
    setSearch(keyFilter);
  };

  const handleSelect = (e) => {
    setDataRender(usedTimeData);
    setType(e.target.value);
  };

  const handleChange = (e) => {
    e.preventDefault();
    setDate(e.target.value);
  };

  return (
    <div className="filter">
      <Row>
        <Col>
          <input
            data-testid="input-name-used"
            className="filter-input"
            onChange={handleSearch}
            placeholder="Search"
            value={search}
          />
        </Col>
        <Col>
          <div className="d-flex input-date">
            <input
              data-testid="input-date-used"
              type="date"
              name="trip-start"
              value={date}
              onChange={handleChange}
            />
          </div>
        </Col>
        <Col>
          <Form.Select
            aria-label="Default select example"
            data-testid="select-box-used"
            onChange={handleSelect}
            value={type}>
            <option>Please select</option>
            {listOS.map((val, index) => {
              return (
                <option key={index + uuidv4()} value={val}>
                  {val}
                </option>
              );
            })}
          </Form.Select>
        </Col>
      </Row>
    </div>
  );
};

export default Filter;
