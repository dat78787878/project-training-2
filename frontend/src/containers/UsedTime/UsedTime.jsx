import React, { useCallback, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import moment from 'moment';
import { getData } from '../../redux/usedTime/actions';
import Loading from '../../components/Loading/Loading';
import { Table } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowUp, faArrowDown } from '@fortawesome/free-solid-svg-icons';
import { v4 as uuidv4 } from 'uuid';
import Menu from '../Menu/Menu';
import Paginatinon from '../Pagination/Paginatinon';

const UsedTime = () => {
  const { usedTimeData, isLoading, isError } = useSelector((state) => state.usedTime);
  const listTitleString = ['OS Name', 'Date'];
  const listTitleNumber = ['Total', 'Facebook', 'Youtube', 'Other'];
  const [dataRender, setDataRender] = useState([]);
  const [page, setPage] = useState(1);
  const [sort, setSort] = useState('');

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getData([page, sort]));
  }, [page, sort]);

  useEffect(() => {
    if (sort == '' || !sort) {
      setDataRender(usedTimeData);
    } else if (sort == 'asc') {
      const dataSortTemp = usedTimeData;
      const dataSort = dataSortTemp.sort((a, b) => a.userName.localeCompare(b.userName));
      setDataRender(dataSort);
    } else {
      const dataSortTemp = usedTimeData;
      const dataSort = dataSortTemp.sort((a, b) => b.userName.localeCompare(a.userName));
      setDataRender(dataSort);
    }
  }, [usedTimeData, sort]);

  useEffect(() => {
    const urlSearchParams = new URLSearchParams(window.location.search);
    const params = Object.fromEntries(urlSearchParams.entries());
    params.page ? '' : (params.page = 1);
    params.sort ? '' : (params.sort = '');
    setPage(params.page);
    setSort(params.sort);
    dispatch(getData([params.page, params.sort]));
  }, [window.location.search]);

  const handleSort = useCallback(
    (s) => {
      if (s === 'asc') {
        setSort('desc');
        const dataSortTemp = usedTimeData;
        const dataSort = dataSortTemp.sort((a, b) => b.userName.localeCompare(a.userName));
        setDataRender(dataSort);
      } else {
        setSort('asc');
        const dataSortTemp = usedTimeData;
        const dataSort = dataSortTemp.sort((a, b) => a.userName.localeCompare(b.userName));
        setDataRender(dataSort);
      }
    },
    [usedTimeData]
  );

  const tableUsedTimeHeader = (
    <thead>
      <tr>
        <th>
          UserName
          {console.log('sort', sort)}
          {sort == 'desc' ? (
            <FontAwesomeIcon
              icon={faArrowDown}
              data-testid="arrow-circle-down"
              onClick={() => handleSort(sort)}
            />
          ) : (
            <FontAwesomeIcon
              icon={faArrowUp}
              data-testid="arrow-circle-up"
              onClick={() => handleSort(sort)}
            />
          )}
        </th>
        {listTitleString.map((val, index) => {
          return <th key={index + uuidv4()}>{val}</th>;
        })}
        {listTitleNumber.map((val, index) => {
          return <th key={index + uuidv4()}>{val}</th>;
        })}
      </tr>
    </thead>
  );
  const tableUsedTime = (
    <>
      <tbody>
        {isLoading ? (
          <tr>
            <td colSpan={7}>
              <Loading />
            </td>
          </tr>
        ) : isError ? (
          <tr>
            <td colSpan={7}>no data</td>
          </tr>
        ) : (
          dataRender.map((val, index) => {
            return (
              <tr key={index + uuidv4()}>
                <td key={index + uuidv4()}>{val.userName}</td>
                <td key={index + uuidv4()}>{val.oSName}</td>
                <td key={index + uuidv4()}>{moment(val.date).format('YYYY-MM-DD')}</td>
                <td key={index + uuidv4()}>{val.useTimeNumber}</td>
                <td key={index + uuidv4()}>{val.facebookTimeUse}</td>
                <td key={index + uuidv4()}>{val.youtubeTimeUse}</td>
                <td key={index + uuidv4()}>{val.other}</td>
              </tr>
            );
          })
        )}
      </tbody>
      <tfoot>
        <tr>
          <td colSpan={7}>
            <Paginatinon page={page} setPage={setPage} sort={sort} />
          </td>
        </tr>
      </tfoot>
    </>
  );

  return (
    <div className="usedTime">
      <Menu />
      <div className="usedTime-container padding-title">
        <Table striped bordered hover>
          {tableUsedTimeHeader}
          {tableUsedTime}
        </Table>
      </div>
    </div>
  );
};

export default UsedTime;
