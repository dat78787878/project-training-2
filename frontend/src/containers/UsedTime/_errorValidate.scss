.modal-error {
    color: rgb(236, 21, 21);
}
