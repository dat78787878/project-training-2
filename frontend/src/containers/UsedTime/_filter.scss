.filter {
  width: 75%;
  padding-bottom: 15px;

  label {
    display: block;
  }

  input,
  label {
    margin: 0.4rem 0;
  }

  .filter-input {
    padding: 5px;
    padding-left: 9px;
    margin: 0 !important;
    width: 250px;
    border-radius: 3.5px;
    border: 1px solid;
  }

  .input-date {
    input {
      margin: 0px;
      padding: 5px;
      padding-left: 8px;
      width: 250px;
      border-radius: 3.5px;
      border: 1px solid;
      margin-left: 4px;
    }
  }

  select {
    border: 1px solid;
    width: 250px;
  }
}
