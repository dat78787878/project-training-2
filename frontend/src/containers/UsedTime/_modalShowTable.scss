.modal-custom-table {
    .modal-content {
        .modal-body {
            input {
                margin: 5px
            }

            .modal-text-date {
                width: 190px;
                padding: 2px;
            }

        }
    }

    select {
        border: 1px solid;
        display: inline-block;
        padding: 3px;
        margin: 5px;
        width: 190px;
    }
   
    .modal-header{
        display: grid;
        .modal-error{
            color: rgb(236, 21, 21);
        }
    }
}

.modal-contain {
    display: flex;

    .modal-item {
        width: 49%;
        margin:5px;
        .modal-text-title {
            font-weight: bold;
            margin-left: 5px;
            font-size: 16px;
        }
    }
}
