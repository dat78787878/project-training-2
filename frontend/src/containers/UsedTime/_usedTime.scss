.usedTime {
  background-color: white;
  height: 530px;

  &-container {
    text-align: center;
    width: 97%;
  }
}

table {
  table-layout: fixed;
  cursor: pointer;

  tbody {
    td {
      vertical-align: middle;
    }
    tr {
      height: 80px;
    }
  }
}

tfoot {
  border-top: none !important;

  tr {
    border: none !important;

    td {
      border: none !important;
    }
  }
}

nav {
  .pagination {
    justify-content: flex-end;
  }

  li a {
    text-decoration: none;
    color: black;
  }
}
