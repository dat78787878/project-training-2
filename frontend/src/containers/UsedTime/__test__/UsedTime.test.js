import React from 'react';
import UsedTime from '../UsedTime';
import { useSelector, useDispatch } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn()
}));

describe('UsedTime', () => {
  let mockAppState = {};
  const mockDispatch = jest.fn();
  beforeEach(() => {
    useDispatch.mockImplementation(() => mockDispatch);
    useSelector.mockImplementation((callback) => {
      return callback(mockAppState);
    });
  });
  afterEach(() => {
    useDispatch.mockClear();
    useSelector.mockClear();
  });
  it('check loading', () => {
    mockAppState = {
      usedTime: {
        usedTimeData: [],
        isLoading: true,
        isError: false
      }
    };
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    expect(screen.getByTestId('loading')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  it('error task', () => {
    mockAppState = {
      usedTime: {
        usedTimeData: [],
        isLoading: false,
        isError: true
      }
    };
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    expect(screen.getByText('no data')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  it('handle Click UsedTime', () => {
    mockAppState = {
      usedTime: {
        usedTimeData: [
          {
            userName: 'pqjx1',
            oSName: 'iOS',
            date: '2016-04-03T13:46:50.804Z',
            useTimeNumber: 11,
            facebookTimeUse: 7,
            youtubeTimeUse: 9,
            other: 18
          },
          {
            userName: 'g5zxl',
            oSName: 'Android',
            date: '2017-04-13T02:05:28.816Z',
            useTimeNumber: 7,
            facebookTimeUse: 6,
            youtubeTimeUse: 8,
            other: 7
          }
        ],
        isLoading: false,
        isError: false
      }
    };
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    const link2 = screen.getByTestId(2);
    fireEvent.click(link2);
    const svg = screen.getByTestId('arrow-circle-up');
    fireEvent.click(svg);
    fireEvent.click(svg);
    const svg1 = screen.getByTestId('arrow-circle-down');
    fireEvent.click(svg1);
    expect(container).toMatchSnapshot();
  });
  it('Filter search', () => {
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    const inputFilter = screen.getByTestId('input-name-used');
    fireEvent.change(inputFilter, { target: { value: '7' } });
    expect(inputFilter.value).toEqual('7');
    fireEvent.change(inputFilter, { target: { value: '' } });
    expect(inputFilter.value).toEqual('');
    expect(container).toMatchSnapshot();
  });
  it('Filter select type', () => {
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    const element = screen.getByTestId('select-box-used');
    fireEvent.click(element);
    fireEvent.change(element, { target: { value: '7' } });
    fireEvent.change(element, { target: { value: 'Please select' } });
    expect(container).toMatchSnapshot();
  });
  it('Filter date', () => {
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    const inputFilter = screen.getByTestId('input-date-used');
    fireEvent.change(inputFilter, { target: { value: '2022-04-13' } });
    expect(inputFilter.value).toEqual('2022-04-13');
    fireEvent.change(inputFilter, { target: { value: '' } });
    expect(inputFilter.value).toEqual('');
    expect(container).toMatchSnapshot();
  });
  it('add element', () => {
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    const inputFilter = screen.getByTestId('add-button');
    fireEvent.click(inputFilter);

    const inputFilter1 = screen.getByTestId('UserNameI');
    fireEvent.change(inputFilter1, { target: { value: '123456789' } });
    expect(inputFilter1.value).toEqual('123456789');

    const inputFilter2 = screen.getByTestId('OSNameI');
    fireEvent.change(inputFilter2, { target: { value: 'iOS' } });
    expect(inputFilter2.value).toEqual('iOS');

    const inputFilter3 = screen.getByTestId('DateI');
    fireEvent.change(inputFilter3, { target: { value: '2022-04-13' } });
    expect(inputFilter3.value).toEqual('2022-04-13');

    const inputFilter4 = screen.getByTestId('YouTubeI');
    fireEvent.change(inputFilter4, { target: { value: '12' } });
    expect(inputFilter4.value).toEqual('12');

    const inputFilter5 = screen.getByTestId('FacebookI');
    fireEvent.change(inputFilter5, { target: { value: '12' } });
    expect(inputFilter5.value).toEqual('12');

    const inputFilter6 = screen.getByTestId('OtherI');
    fireEvent.change(inputFilter6, { target: { value: '12' } });
    expect(inputFilter6.value).toEqual('12');

    const inputFilter7 = screen.getByTestId('btnOK');
    fireEvent.click(inputFilter7);

    expect(container).toMatchSnapshot();
  });

  it('edit element', () => {
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    const inputFilter = screen.getByTestId('edit-button0');
    fireEvent.click(inputFilter);

    const inputFilter1 = screen.getByTestId('UserNameI');
    fireEvent.change(inputFilter1, { target: { value: '123456789' } });
    expect(inputFilter1.value).toEqual('123456789');

    const inputFilter3 = screen.getByTestId('DateI');
    fireEvent.change(inputFilter3, { target: { value: '2022-04-13' } });
    expect(inputFilter3.value).toEqual('2022-04-13');

    const inputFilter4 = screen.getByTestId('YouTubeI');
    fireEvent.change(inputFilter4, { target: { value: '12' } });
    expect(inputFilter4.value).toEqual('12');

    const inputFilter5 = screen.getByTestId('FacebookI');
    fireEvent.change(inputFilter5, { target: { value: '12' } });
    expect(inputFilter5.value).toEqual('12');

    const inputFilter6 = screen.getByTestId('OtherI');
    fireEvent.change(inputFilter6, { target: { value: '12' } });
    expect(inputFilter6.value).toEqual('12');

    const inputFilter7 = screen.getByTestId('btnOK');
    fireEvent.click(inputFilter7);

    const inputFilter8 = screen.getByTestId('btnCancel');
    fireEvent.click(inputFilter8);
    expect(container).toMatchSnapshot();
  });

  it('error input element', () => {
    const { container } = render(
      <BrowserRouter>
        <UsedTime />
      </BrowserRouter>
    );
    const inputFilter = screen.getByTestId('add-button');
    fireEvent.click(inputFilter);

    const inputFilter1 = screen.getByTestId('UserNameI');
    fireEvent.change(inputFilter1, { target: { value: '' } });
    expect(inputFilter1.value).toEqual('');

    const inputFilter2 = screen.getByTestId('OSNameI');
    fireEvent.change(inputFilter2, { target: { value: '' } });
    expect(inputFilter2.value).toEqual('Please select');

    const inputFilter3 = screen.getByTestId('DateI');
    fireEvent.change(inputFilter3, { target: { value: '' } });
    expect(inputFilter3.value).toEqual('');

    const inputFilter4 = screen.getByTestId('YouTubeI');
    fireEvent.change(inputFilter4, { target: { value: '' } });
    expect(inputFilter4.value).toEqual('');

    const inputFilter5 = screen.getByTestId('FacebookI');
    fireEvent.change(inputFilter5, { target: { value: '' } });
    expect(inputFilter5.value).toEqual('');

    const inputFilter6 = screen.getByTestId('OtherI');
    fireEvent.change(inputFilter6, { target: { value: '' } });
    expect(inputFilter6.value).toEqual('');

    const inputFilter7 = screen.getByTestId('btnOK');
    fireEvent.click(inputFilter7);

    expect(container).toMatchSnapshot();
  });
});
