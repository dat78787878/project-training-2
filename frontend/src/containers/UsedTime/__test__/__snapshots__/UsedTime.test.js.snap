// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsedTime check loading 1`] = `
<div>
  <div
    class="usedTime"
  >
    <div
      class="menu"
    >
      <nav
        id="nav"
        role="navigation"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Anlysis
            </a>
          </li>
          <li>
            <a
              class=""
              href="/used_time"
            >
              Used time
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="usedTime-container padding-title"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              UserName
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-up "
                data-icon="arrow-up"
                data-prefix="fas"
                data-testid="arrow-circle-up"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M374.6 246.6C368.4 252.9 360.2 256 352 256s-16.38-3.125-22.62-9.375L224 141.3V448c0 17.69-14.33 31.1-31.1 31.1S160 465.7 160 448V141.3L54.63 246.6c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0l160 160C387.1 213.9 387.1 234.1 374.6 246.6z"
                  fill="currentColor"
                />
              </svg>
            </th>
            <th>
              OS Name
            </th>
            <th>
              Date
            </th>
            <th>
              Total
            </th>
            <th>
              Facebook
            </th>
            <th>
              Youtube
            </th>
            <th>
              Other
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              colspan="7"
            >
              <div
                class="loading"
              >
                <div
                  class="spinner-border text-success"
                  data-testid="loading"
                />
              </div>
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colspan="7"
            >
              <div
                class="usedTime-container-pagination"
              >
                
                <nav
                  aria-label="Countries Pagination"
                >
                  <ul
                    class="pagination"
                  >
                    <li
                      class="page-item active"
                    >
                      <a
                        href="/used_time?page=1&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="1"
                          href="#"
                        >
                          1
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=2&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="2"
                          href="#"
                        >
                          2
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=3&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="3"
                          href="#"
                        >
                          3
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=4&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="4"
                          href="#"
                        >
                          4
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=5&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="5"
                          href="#"
                        >
                          5
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=2&sort="
                      >
                        <a
                          aria-expanded="false"
                          aria-label="Next"
                          class="page-link"
                          href="#"
                        >
                          <span
                            aria-hidden="true"
                          >
                            Â»
                          </span>
                          <span
                            class="sr-only"
                          >
                            Next
                          </span>
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=10&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="10"
                          href="#"
                        >
                          10
                        </a>
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;

exports[`UsedTime error task 1`] = `
<div>
  <div
    class="usedTime"
  >
    <div
      class="menu"
    >
      <nav
        id="nav"
        role="navigation"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Anlysis
            </a>
          </li>
          <li>
            <a
              class=""
              href="/used_time"
            >
              Used time
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="usedTime-container padding-title"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              UserName
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-up "
                data-icon="arrow-up"
                data-prefix="fas"
                data-testid="arrow-circle-up"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M374.6 246.6C368.4 252.9 360.2 256 352 256s-16.38-3.125-22.62-9.375L224 141.3V448c0 17.69-14.33 31.1-31.1 31.1S160 465.7 160 448V141.3L54.63 246.6c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0l160 160C387.1 213.9 387.1 234.1 374.6 246.6z"
                  fill="currentColor"
                />
              </svg>
            </th>
            <th>
              OS Name
            </th>
            <th>
              Date
            </th>
            <th>
              Total
            </th>
            <th>
              Facebook
            </th>
            <th>
              Youtube
            </th>
            <th>
              Other
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              colspan="7"
            >
              no data
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colspan="7"
            >
              <div
                class="usedTime-container-pagination"
              >
                
                <nav
                  aria-label="Countries Pagination"
                >
                  <ul
                    class="pagination"
                  >
                    <li
                      class="page-item active"
                    >
                      <a
                        href="/used_time?page=1&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="1"
                          href="#"
                        >
                          1
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=2&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="2"
                          href="#"
                        >
                          2
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=3&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="3"
                          href="#"
                        >
                          3
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=4&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="4"
                          href="#"
                        >
                          4
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=5&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="5"
                          href="#"
                        >
                          5
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=2&sort="
                      >
                        <a
                          aria-expanded="false"
                          aria-label="Next"
                          class="page-link"
                          href="#"
                        >
                          <span
                            aria-hidden="true"
                          >
                            Â»
                          </span>
                          <span
                            class="sr-only"
                          >
                            Next
                          </span>
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=10&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="10"
                          href="#"
                        >
                          10
                        </a>
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;

exports[`UsedTime handle Click UsedTime 1`] = `
<div>
  <div
    class="usedTime"
  >
    <div
      class="menu"
    >
      <nav
        id="nav"
        role="navigation"
      >
        <ul>
          <li>
            <a
              class=""
              href="/"
            >
              Anlysis
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/used_time"
            >
              Used time
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="usedTime-container padding-title"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              UserName
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-up "
                data-icon="arrow-up"
                data-prefix="fas"
                data-testid="arrow-circle-up"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M374.6 246.6C368.4 252.9 360.2 256 352 256s-16.38-3.125-22.62-9.375L224 141.3V448c0 17.69-14.33 31.1-31.1 31.1S160 465.7 160 448V141.3L54.63 246.6c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0l160 160C387.1 213.9 387.1 234.1 374.6 246.6z"
                  fill="currentColor"
                />
              </svg>
            </th>
            <th>
              OS Name
            </th>
            <th>
              Date
            </th>
            <th>
              Total
            </th>
            <th>
              Facebook
            </th>
            <th>
              Youtube
            </th>
            <th>
              Other
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              g5zxl
            </td>
            <td>
              Android
            </td>
            <td>
              2017-04-13
            </td>
            <td>
              7
            </td>
            <td>
              6
            </td>
            <td>
              8
            </td>
            <td>
              7
            </td>
          </tr>
          <tr>
            <td>
              pqjx1
            </td>
            <td>
              iOS
            </td>
            <td>
              2016-04-03
            </td>
            <td>
              11
            </td>
            <td>
              7
            </td>
            <td>
              9
            </td>
            <td>
              18
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colspan="7"
            >
              <div
                class="usedTime-container-pagination"
              >
                
                <nav
                  aria-label="Countries Pagination"
                >
                  <ul
                    class="pagination"
                  >
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=1&sort=asc"
                      >
                        <a
                          class="page-link"
                          data-testid="1"
                          href="#"
                        >
                          1
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=8&sort=asc"
                      >
                        <a
                          aria-label="Previous"
                          class="page-link"
                          href="#"
                        >
                          <span
                            aria-hidden="true"
                          >
                            Â«
                          </span>
                          <span
                            class="sr-only"
                          >
                            Previous
                          </span>
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=6&sort=asc"
                      >
                        <a
                          class="page-link"
                          data-testid="6"
                          href="#"
                        >
                          6
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=7&sort=asc"
                      >
                        <a
                          class="page-link"
                          data-testid="7"
                          href="#"
                        >
                          7
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=8&sort=asc"
                      >
                        <a
                          class="page-link"
                          data-testid="8"
                          href="#"
                        >
                          8
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item active"
                    >
                      <a
                        href="/used_time?page=9&sort=asc"
                      >
                        <a
                          class="page-link"
                          data-testid="9"
                          href="#"
                        >
                          9
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=10&sort=asc"
                      >
                        <a
                          class="page-link"
                          data-testid="10"
                          href="#"
                        >
                          10
                        </a>
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;

exports[`UsedTime render UsedTime 1`] = `
<div>
  <div
    class="usedTime"
  >
    <div
      class="menu"
    >
      <nav
        id="nav"
        role="navigation"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Anlysis
            </a>
          </li>
          <li>
            <a
              class=""
              href="/used_time"
            >
              Used time
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="usedTime-container padding-title"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              UserName
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-up "
                data-icon="arrow-up"
                data-prefix="fas"
                data-testid="arrow-circle-up"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M374.6 246.6C368.4 252.9 360.2 256 352 256s-16.38-3.125-22.62-9.375L224 141.3V448c0 17.69-14.33 31.1-31.1 31.1S160 465.7 160 448V141.3L54.63 246.6c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0l160 160C387.1 213.9 387.1 234.1 374.6 246.6z"
                  fill="currentColor"
                />
              </svg>
            </th>
            <th>
              OS Name
            </th>
            <th>
              Date
            </th>
            <th>
              Total
            </th>
            <th>
              Facebook
            </th>
            <th>
              Youtube
            </th>
            <th>
              Other
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              g5zxl
            </td>
            <td>
              Android
            </td>
            <td>
              2017-04-13
            </td>
            <td>
              7
            </td>
            <td>
              6
            </td>
            <td>
              8
            </td>
            <td>
              7
            </td>
          </tr>
          <tr>
            <td>
              pqjx1
            </td>
            <td>
              iOS
            </td>
            <td>
              2016-04-03
            </td>
            <td>
              11
            </td>
            <td>
              7
            </td>
            <td>
              9
            </td>
            <td>
              18
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colspan="7"
            >
              <div
                class="usedTime-container-pagination"
              >
                
                <nav
                  aria-label="Countries Pagination"
                >
                  <ul
                    class="pagination"
                  >
                    <li
                      class="page-item active"
                    >
                      <a
                        href="/used_time?page=1&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="1"
                          href="#"
                        >
                          1
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=2&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="2"
                          href="#"
                        >
                          2
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=3&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="3"
                          href="#"
                        >
                          3
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=4&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="4"
                          href="#"
                        >
                          4
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=5&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="5"
                          href="#"
                        >
                          5
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=2&sort="
                      >
                        <a
                          aria-expanded="false"
                          aria-label="Next"
                          class="page-link"
                          href="#"
                        >
                          <span
                            aria-hidden="true"
                          >
                            Â»
                          </span>
                          <span
                            class="sr-only"
                          >
                            Next
                          </span>
                        </a>
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        href="/used_time?page=10&sort="
                      >
                        <a
                          class="page-link"
                          data-testid="10"
                          href="#"
                        >
                          10
                        </a>
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
